### PR TITLE
Move required internal/core/internal components to internal/coreinternal

### PR DESCRIPTION
**Pull request by alexperez52**
**Description:** <Describe what has changed.>
This moves the following components from internal/core/internal into internal/coreinternal
- internal/testdata/
- internal/processor/
- internal/sharedcomponent/
- internal/internalconsumertest/
- internal/goldendataset/
- internal/idutils/

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/3474

### CODE REVIEW

File: internal/coreinternal/go.mod [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+go 1.16
+
+require (
+	github.com/antonmedv/expr v1.9.0
+	github.com/golang/groupcache v0.0.0-20210331224755-41bb18bfe9da
+	github.com/spf13/cast v1.4.1
+	github.com/stretchr/testify v1.7.0
+	go.opentelemetry.io/collector v0.33.0
+	go.opentelemetry.io/collector/model v0.33.0
```
Comment (bogdandrutu) [inline internal/coreinternal/go.mod:11]: Use version `v0.33.1-0.20210820002854-d3000232f8f6` as we do everywhere

**Review by bogdandrutu — COMMENTED** [review]
Add a Makefile like https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/extension/httpforwarder/Makefile

File: internal/coreinternal/Makefile [hunk 2]
```diff
@@ -0,0 +1 @@
+include ../../Makefile.Common
```
Comment (bogdandrutu) [inline internal/coreinternal/Makefile:1]: New line :D

Comment (alexperez52) [reply internal/coreinternal/Makefile]: Added

**Review by bogdandrutu — APPROVED** [review]

**Comment by bogdandrutu** [comment]
```
processor/filterset/config_test.go:24: File is not `goimports`-ed with -local github.com/open-telemetry/opentelemetry-collector-contrib (goimports)
	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterset/regexp"
processor/filtermatcher/attributematcher.go:21: File is not `goimports`-ed with -local github.com/open-telemetry/opentelemetry-collector-contrib (goimports)

processor/filtermatcher/filtermatcher.go:19: File is not `goimports`-ed with -local github.com/open-telemetry/opentelemetry-collector-contrib (goimports)

```

File: internal/coreinternal/processor/filterspan/filterspan.go [hunk 3]
```diff
@@ -17,9 +17,9 @@ package filterspan
...
 	"fmt"
 
-	"go.opentelemetry.io/collector/internal/processor/filterconfig"
-	"go.opentelemetry.io/collector/internal/processor/filtermatcher"
-	"go.opentelemetry.io/collector/internal/processor/filterset"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterconfig"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filtermatcher"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterset"
 	"go.opentelemetry.io/collector/model/pdata"
 	conventions "go.opentelemetry.io/collector/translator/conventions/v1.5.0"
```
Comment (bogdandrutu) [inline internal/coreinternal/processor/filterspan/filterspan.go:24]:
```

import (
	"fmt"

	"go.opentelemetry.io/collector/model/pdata"
	conventions "go.opentelemetry.io/collector/translator/conventions/v1.5.0"

	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterconfig"
	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filtermatcher"
	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/processor/filterset"
)
```

**Pull request merged**